### PR TITLE
feat: username validation for length and special chars

### PR DESCRIPTION
**Pull request by alexandra-marin**
#### Relevant info and issue/PR links 
 https://app.clubhouse.io/peerio/story/12748/sdk-username-validation-does-not-check-for-length-specifically
https://app.clubhouse.io/peerio/story/12772/sdk-username-suggestions-don-t-work-if-name-has-hyphen
#### Testing instructions  

----
### Repository owner

- [ ] Was tested and can be merged.
- [ ] PR name follows conventional changelog format.

[PR guideline](https://github.com/PeerioTechnologies/peerio-icebear/blob/dev/docs/CONTRIBUTING.md)

### CODE REVIEW

File: src/helpers/validation/user-validators.js [hunk 1]
```diff
@@ -71,6 +72,13 @@ function _callServer(context, name, value) {
     });
 }
 
+function isValidUsernameLength(name) {
+    if (name) {
+        return Promise.resolve(name.length < usernameLength);
```
Comment (anri-asaturov) [inline src/helpers/validation/user-validators.js]: <=? currently peerio supports 16char usernames

File: src/helpers/validation/user-validators.js [hunk 2]
```diff
@@ -159,7 +164,14 @@ const suggestUsername = async (firstName, lastName) => {
         `${firstName}_${lastName}${lastName}`
     ];
 
-    for (const option of options) {
+    const validOptions = options.map(x => x
+        .trim()
+        .replace(/[^a-z|A-Z|0-9|_]/g, '')
+        .substring(0, usernameLength - 1)
```
Comment (anri-asaturov) [inline src/helpers/validation/user-validators.js]: again, if maximum is 16 chars, then remove `-1`

**Comment by anri-asaturov** [comment]
run pretttier, tests are failing (i recommend vscode plugin)

File: src/helpers/validation/user-validators.js (hunk 2 above)
Comment (alexandra-marin) [reply src/helpers/validation/user-validators.js]: if i want 16 chars, don't i substring 0 to 15?

Comment (anri-asaturov) [reply src/helpers/validation/user-validators.js]:
```'1111111111111111111111111111111'.substring(0, 15).length
15```

btw copy file is conflicting now

File: src/copy/en.json [hunk 3]
```diff
@@ -213,7 +213,7 @@
   "error_fieldRequired": "Required",
   "error_usernameBadFormat": "Only letters, numbers or underscore allowed.",
   "error_usernameLengthExceeded": "Usernames must have less than 16 characters",
```
Comment (anri-asaturov) [inline src/copy/en.json]: This copy seems incorrect, username can be 16 characters length

**Comment by ebilg** [comment]
can be merged

**Pull request merged**